docs/guides/conda.rst

```diff
@@ -8,16 +8,171 @@ and need them installed when building their documentation.
 
 This work was funded by `Clinical Graphics`_ -- many thanks for their support of Open Source.
 
+.. _Clinical Graphics: https://www.clinicalgraphics.com/
+
 Activating Conda
 ----------------
 
 Conda support is available using a :doc:`../config-file/index`, see :ref:`config-file/v2:conda`.
 
-This Conda environment will also have Sphinx and other build time dependencies installed.
-It will use the same order of operations that we support currently:
+Our :ref:`Docker images <builds:Docker images>` use Miniconda, a minimal conda installer.
+After specifying your project requirements using conda ``environment.yml`` files,
+Read the Docs will create the environment (using ``conda env create``)
+and add the core dependencies needed to build the documentation.
+
+Creating the ``environment.yml``
+--------------------------------
+
+There are several ways of `exporting a conda environment`_:
+
+- ``conda env export`` will produce a complete list of all the packages installed in the environment
+  with their exact versions. This is the best option to ensure reproducibility,
+  but can create problems if done from a different operative system than the target machine.
+- ``conda env export --from-history`` will only include packages that were explicitly requested,
+  excluding the transitive dependencies. This is the best option to maximize cross-platform compatibility.
+- And finally, you can also write it by hand. This allows for more fine-tuned control and overcomes
+  some limitations in the conda exporting capabilities.
+
+For example, using the second method for an existing environment:
+
+.. code-block::
+
+    $ conda activate rtd38
+    (rtd38) $ conda env export --from-history | tee environment.yml
+    name: rtd38
+    channels:
+      - defaults
+      - conda-forge
+    dependencies:
+      - rasterio==1.2
+      - python=3.8
+      - pytorch-cpu=1.7
+    prefix: /home/juanlu/.mambaforge/envs/rtd38
+
+Read the Docs will override the ``name`` and ``prefix`` of the environment when creating it,
+so they can have any value, or not be present at all.
+
+.. tip:: Bear in mind that ``rasterio==1.2`` will install version ``1.2.0``,
+   whereas ``python=3.8`` will fetch the latest ``3.8.*``,
+   which is ``3.8.8`` at the time of writing.
+
+.. warning:: Pinning Sphinx and other Read the Docs core dependencies
+   is not yet supported by default when using conda (see `this GitHub issue for discussion`_).
+   If your project needs it, request that we enable the ``CONDA_APPEND_CORE_REQUIREMENTS``
+   :ref:`feature flag <guides/feature-flags:Feature Flags>`.
```

Comment:
I hope we find a better solution soon. This is bad UX :disappointed: --maybe it worth to enable this feature by default on new projects and start testing it more? Besides fixing this issue it also speeds up the creation/installation of packages.

Same here; I think it's out of the scope of this PR, but I'm happy that we have a note in this document mentioning as a workaround.

Reply:
+1 on testing it more to have it enable by default

Reply:
Agreed. We should make it the default -- perhaps `future_default_true` for now? Then see if we can backport it?